pkg/kubectl/cmd/set/set_port.go

```diff
@@ -0,0 +1,281 @@
+	Encoder           runtime.Encoder
+	Selector          string
+	Out               io.Writer
+	Err               io.Writer
+	DryRun            bool
+	ShortOutput       bool
+	All               bool
+	Record            bool
+	Output            string
+	ChangeCause       string
+	Local             bool
+	Cmd               *cobra.Command
+	port              int
+	protocol          string
+	overwrite         bool
+	containerSelector string
+	portsForObject    func(object runtime.Object) ([]string, error)
+
+	PrintObject            func(cmd *cobra.Command, isLocal bool, mapper meta.RESTMapper, obj runtime.Object, out io.Writer) error
+	UpdatePodSpecForObject func(obj runtime.Object, fn func(*api.PodSpec) error) (bool, error)
+}
+
+var (
+	portResources = `
+  	pod (po), replicationcontroller (rc), deployment (deploy), daemonset (ds), job, replicaset (rs)`
+
+	portLong = templates.LongDesc(`
+		Update existing container port(s) of resources.
+
+		Possible resources include (case insensitive):
+		` + portResources)
+
+	portExample = templates.Examples(`
+		# Set a deployment's nginx container port to 80.
+		kubectl set port deployment/nginx --port=80 -c=nginx
+
+		# Update all deployments' and rc's nginx container's image to 'nginx:1.9.1'
+		kubectl set port deployments,rc --port=80 --all
+
+		# Update port of all containers of daemonset abc to 'nginx:1.9.1'
+		kubectl set port daemonset abc --port=80
+
+		# Print result (in yaml format) of updating container port from local file, without hitting the server
+		kubectl set port -f path/to/file.yaml --port=80 --local -o yaml`)
+)
+
+func NewCmdPort(f cmdutil.Factory, out, err io.Writer) *cobra.Command {
+	options := &PortOptions{
+		Out: out,
+		Err: err,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "port (-f FILENAME | TYPE NAME)",
+		Short:   i18n.T("Update port of a pod template"),
+		Long:    portLong,
+		Example: portExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(options.Complete(f, cmd, args))
+			cmdutil.CheckErr(options.Validate())
+			cmdutil.CheckErr(options.Run())
+		},
+	}
+
+	cmdutil.AddPrinterFlags(cmd)
+	usage := "identifying the resource to get from a server."
+	cmdutil.AddFilenameOptionFlags(cmd, &options.FilenameOptions, usage)
+	cmd.Flags().StringVarP(&options.containerSelector, "containers", "c", "*", "The names of containers in the selected pod templates to change - may use wildcards")
+	cmd.Flags().BoolVar(&options.All, "all", false, "Select all resources in the namespace of the specified resource types")
+	cmd.Flags().StringVarP(&options.Selector, "selector", "l", "", "Selector (label query) to filter on, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2)")
+	cmd.Flags().BoolVar(&options.Local, "local", false, "If true, set port will NOT contact api-server but run locally.")
+	cmd.Flags().Int("port", 80, i18n.T("Name or number for the port on the container that the service should direct traffic to. Optional."))
```

Comment:
`port` should be marked as required? I think we'd better not set default value, especially `80`. WDYT?

Reply:
yes, i will updated